pages/map.py

```diff
@@ -244,33 +312,39 @@ def update_output(map_type, selected_user_ids, selected_user_emails, trips_data)
     Output('user-id-dropdown', 'disabled'),
     Output('user-email-dropdown', 'disabled'),
     Input('map-type-dropdown', 'value'),
+    Input('user-mode-dropdown', 'value'),
 )
-def control_user_dropdowns(map_type):
+def control_user_dropdowns(map_type,selected_user_modes):
     disabled = True
     if map_type == 'lines':
         disabled = False
+        if selected_user_modes:
+            disabled = True
     return disabled, disabled
 
 
+def process_trips_group(trips_group):
+    users_data = dict()
+    if trips_group:
+        keys = list(trips_group)
+        n = len(keys) % 360
+        k = 359 // (n - 1) if n > 1 else 0
+        for ind, key in enumerate(trips_group.groups.keys()):
```

Comment:
Can you please add the explanation

> This code block was previously inside store_trips_map_data (it processes a group of trips, assigns color to each group, sorts the trips within each group and stores the processed data in a dictionary).

to the code?

Not everybody is going to look at the related PR, especially when the information is hidden in a detail